experimenter/experimenter/klaatu/client.py

```diff
@@ -31,65 +88,83 @@ def run_test(self, experiment_slug, branch_slug, targets):
 
         response = requests.post(url, headers=self.headers, json=data)
         if response.status_code != 204:
-            raise Exception(
-                f"Failed to trigger workflow: {response.status_code}, {response.text}"
+            raise KlaatuError(
+                "Failed to trigger workflow",
+                status_code=response.status_code,
+                response_text=response.text,
             )
 
-    def find_run_id(self, experiment_slug, dispatched_at=None):
-        created_filter = ""
+    def find_run_id(
+        self, experiment_slug: str, dispatched_at: Optional[datetime.datetime] = None
+    ) -> int:
+        query = {"event": "workflow_dispatch"}
         if dispatched_at:
-            created_filter = f"&created=>={dispatched_at.isoformat()}Z"
+            query["created"] = f">={dispatched_at.isoformat()}Z"
 
-        url = (
-            f"{self.base_url}/actions/workflows/{self.workflow_name}/runs"
-            f"?event=workflow_dispatch{created_filter}"
-        )
+        path = KlaatuEndpoints.RUNS.format(workflow=self.workflow_name)
+        base_url = urljoin(self.base_url, path)
+        url = base_url + "?" + urlencode(query)
```

Comment:
Could we do this as an f-string?

Reply:
Oh in general yes, always f string, but I was saying to @RJAK11 the other day that specifically when joining urls or file paths, it's better to use the 'urljoin' thingy cuz it'll catch edge cases with trailing/leading slahes, etc

Reply:
Oh yeah, didn't even see this was a URL.